cli/src/command/account.rs

```diff
@@ -919,40 +919,68 @@ async fn print_address(account: &Account, address: &AccountAddress) -> Result<()
     let addresses = account.addresses_with_unspent_outputs().await?;
     let current_time = iota_sdk::utils::unix_timestamp_now().as_secs() as u32;
 
+    let mut output_ids = &Vec::new();
+    let mut address_amount = 0;
+    let mut address_nts = Vec::<TokenId>::new();
+    let mut address_nfts = Vec::new();
+    let mut address_aliases = Vec::new();
+    let mut address_foundries = Vec::new();
+
     if let Ok(index) = addresses.binary_search_by_key(&(address.key_index(), address.internal()), |a| {
         (a.key_index(), a.internal())
     }) {
-        let mut address_amount = 0;
-        for output_id in addresses[index].output_ids() {
+        output_ids = addresses[index].output_ids();
+
+        for output_id in output_ids {
             if let Some(output_data) = account.get_output(output_id).await {
-                // Output might be associated with the address, but can't unlocked by it, so we check that here
+                // Output might be associated with the address, but can't be unlocked by it, so we check that here
+                // TODO: Should we just skip outputs for which this call fails? (that's only possible for
+                // TreasuryOutputs, right?)
                 let (required_address, _) =
                     output_data
                         .output
                         .required_and_unlocked_address(current_time, output_id, None)?;
-                if *address.address().as_ref() == required_address {
+
+                if address.address().as_ref() == &required_address {
+                    if let Some(nts) = output_data.output.native_tokens() {
+                        address_nts.extend(nts.iter().map(|nt| nt.token_id()));
+                    }
+                    if output_data.output.is_nft() {
+                        address_nfts.push(output_data.output.as_nft().nft_id_non_null(&output_id));
+                    }
+                    if output_data.output.is_alias() {
+                        address_aliases.push(output_data.output.as_alias().alias_id_non_null(&output_id));
+                    }
+                    if output_data.output.is_foundry() {
+                        address_foundries.push(output_data.output.as_foundry().id());
+                    }
```

Comment:
if is_ + as_ sounds like a match...

Reply:
or if let

Reply:
normally it does, yes, but the `as_` methods panic if you call them on the wrong output type.

Reply:
Yeah I know that, but why not match ?

Reply:
dunno ... I found it readable enough I guess.

Reply:
changing it ...

Reply:
Yeah but it doesn't scale, we're 100% going to forget about adding delegation output in 2.0

Reply:
replaced it as suggested with a match